### PR TITLE
python: Move some of the testing code for dazl into dazl itself

### PR DESCRIPTION
**Pull request by da-tanabe**
python: Move some of the testing code for dazl into dazl itself so that it can be used by clients as well.

### CODE REVIEW

**Review by maxdeliso-da — APPROVED** [review]
```
  , _ ,
 ( o o )
/'` ' `'\
|'''''''|
|\\'''//|
   """
```


```
  , _ ,
 ( o o )
/'` ' `'\
|'''''''|
|\\'''//|
   """
```




```
  , _ ,
 ( o o )
/'` ' `'\
|'''''''|
|\\'''//|
   """
```

File: docs/dazl.testing.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+   # Python 3.7+ only
+   asyncio.run(main())
+
+
+.. py:function:: sandbox
+
+   :param project_root:
+      The root project directory for your Daml project. If supplied, the DAR from this project is
+      automatically uploaded that ``daml start`` does. If ``None``, an empty sandbox is started and
```
Comment (sammy-da) [inline docs/dazl.testing.rst:41]: automatically uploaded ~that~ **__like__** ``daml start`` does?

File: python/dazl/testing/_sandbox.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                    "start",
+                    "--start-navigator=no",
+                    "--open-browser=no",
+                    f"--sandbox-port={port}",
+                ]
+                cwd = self._project_root
+            else:
+                cmdline = ["daml", "sandbox", "--port", str(port)]
+
+            self._process = subprocess.Popen(
```
Comment (maxdeliso-da) [inline python/dazl/testing/_sandbox.py:94]: no :mario-pipe: 😢

File: docs/dazl.testing.rst (hunk 1 above)
Comment (da-tanabe) [reply docs/dazl.testing.rst:41]: oops thx

File: docs/testing.rst [hunk 3]
```diff
@@ -41,26 +41,19 @@ globals:
...
-        port = dazl.util.find_free_port()
-
-        sandbox_proc = subprocess.Popen(
-            ["daml", "start", "--start-navigator=no", "--open-browser=no", f"--sandbox-port={port}"],
-            cwd=daml_project_root)
-        try:
-            yield "http://localhost:{port}"
-        finally:
-            sandbox_proc.terminate()
+        with dazl.testing.sandbox(project_root=daml_project_root) as sandbox_proc:
```
Comment (mschaef-da) [inline docs/testing.rst:44]: Good to see this back.

**Review by mschaef-da — APPROVED** [review]

**Comment by mschaef-da** [comment]
Oh... almost forgot this. 
image

**Review by sammy-da — APPROVED** [review]

File: docs/testing.rst (hunk 3 above)
Comment (da-tanabe) [reply docs/testing.rst:44]: heh yep…

And in this form, I think it threads the needle between being relatively easy to maintain and also not so sensitive to Daml Connect releases, which is what made the code removed from #98 a bit more distasteful.

**Pull request merged**